server/utils/queryHelpers/communitySanitize.js

```diff
@@ -35,5 +23,39 @@ export default (communityData, locationData, loginData, scopeData) => {
 				page: availablePages[collection.pageId],
 			};
 		});
+
+	cleanedData.pages = cleanedData.pages.filter((item) => {
```

Comment:
I moved the page filter below the collection sanitation logic to only consider collections where the user has explicit permission when checking if a page should be visible due to collection-level permissions.